lib/index.js

```diff
@@ -0,0 +1,48 @@
+const shortid = require('shortid');
+
+class MemoryDatabase {
+  constructor() {
+    this.store = {};
+  }
+
+  create(obj) {
+    const _id = shortid.generate();
+    const copiedObj = { ...obj, _id };
+    this.store[_id] = copiedObj;
+    return copiedObj;
+  }
+  findById(_id) {
+    const obj = this.store[_id];
+    if (!obj) {
+      throw `No object with _id ${_id}`;
+    }
+    return obj;
+  }
+  find() {
+    return Object.values(this.store);
+  }
+  findByIdAndUpdate(_id, newObj) {
+    if (!this.exists(_id)) {
```

Comment:
```suggestion
    if (!this.store[_id]) {
```
This is another way to check if the id exists in the store.